crates/daphne/src/roles/leader/mod.rs

```diff
@@ -223,6 +226,34 @@ pub async fn handle_upload_req<A: DapLeader>(
         .into());
     }
 
+    if let Some(public_extensions) = &report.report_metadata.public_extensions {
+        // We can be sure at this point that the ReportMetadata is well formed (as
+        // the decoding / error checking happens in the extractor).
+        assert_eq!(DapVersion::Latest, task_config.version);
+        let mut unknown_extensions = Vec::<u16>::new();
+        if protocol::check_no_duplicates(public_extensions.iter()).is_err() {
+            return Err(DapError::Abort(DapAbort::InvalidMessage {
+                detail: "Repeated public extension".into(),
+                task_id,
+            }));
+        };
```

Comment:
IF the metdata "well-formed" (see above comment), then do we need to check for repeats here?

Reply:
Well formed in the sense that if the version is `DapVersion::Latest` then `public_extensions.is_some()`, and that if the version is `DapVersion::Draft09` then `public_extensions.is_none()`. It doesn't say the data is good, just that it's correctly formed.